File: chapters/09_multi_agent_routing/v1_single_agent.ts
// TypeScript mirror for Chapter 09: Multi-Agent Routing (v1)
// Run with: npx ts-node v1_single_agent.ts

function runSingleAgent(task: string): { agent: string; task: string; result: string } {
  return {
    agent: "generalist",
    task,
    result: `generalist handled: ${task}`,
  };
}

function main(): void {
  console.log(runSingleAgent("build ui"));
}

main();

File: chapters/09_multi_agent_routing/v1_single_agent.py
from __future__ import annotations


def run_single_agent(task: str) -> dict[str, str]:
    return {
        "agent": "generalist",
        "task": task,
        "result": f"generalist handled: {task}",
    }


def main() -> None:
    print(run_single_agent("build ui"))


if __name__ == "__main__":
    main()
